authentication/tests/test_permissions.py

```diff
@@ -0,0 +1,53 @@
+from django.test import TestCase
+from django.contrib.auth.models import User, Group
+from emis.permissions import EmisPermission, EmisPermMode, decompose_perm_code, get_code
+from emis.groups import PERMISSIONS_BY_GROUP
+
+
+class GroupPermissionTests(TestCase):
+    @staticmethod
+    def create_user_in_group(group_name: str) -> User:
```

Comment:
This seems to me like a userful method that should exist outside this test class. 

The flaw I see is some of the tests below are literally just testing this method works as expected:

test_user = GroupPermissionTests.create_user_in_group(group_name)

They are not really testing something outside the test class works as expected. 

We should try to follow the same flow that the "user of the application" would and test that flow. Not this method.

Reply:
Moved out of class into helpers.  I agree tests which are more representative of the flow should be added.  The existing tests so far serve two purposes:
- Verify that if a user is assigned to a pre-built/migrated group, that all expected permissions for the group are available to the user.
- Verify permissions module initialization works allows the full "<app>.<permission code name>" string to be obtained via EmisPermission.get_create_code, get_update_code, etc.